module/spl/spl-kmem-cache.c

```diff
@@ -1213,6 +1225,10 @@ spl_cache_grow(spl_kmem_cache_t *skc, int flags, void **obj)
 		return (rc ? rc : -EAGAIN);
 	}
 
+	rc = __spl_cache_grow(skc, flags | KM_NOSLEEP);
+	if (rc == 0)
+		return (rc);
```

Comment:
It's functionally identical but I think a `return (0)` here would flow better.  Also please add a comment describing the motivation behind adding this KM_NOSLEEP allocation.